### PR TITLE
TILA-2074 | Reservation staff create mutation

### PR DESCRIPTION
**Pull request by nicobav**
## Change log
- Refactors reservation serializers to own sub package and files.
- Introduce new mutation createStaffReservation which is intended for staff users to create reservations

## Other notes
Happy reviewing 🥂

### CODE REVIEW

File: api/graphql/reservations/reservation_serializers/staff_create_serializers.py [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+        self.set_buffer_times(data)
+
+        user = self.context.get("request").user
+        data["user"] = user
+
+        return data
+
+    def set_buffer_times(self, data):
+        if not data.get("buffer_time_after"):
+            data["buffer_time_after"] = None
```
Comment (saaste) [inline api/graphql/reservations/reservation_serializers/staff_create_serializers.py:179]: Could you clarify what is the purpose of this? Initially it looked like you are trying to set `buffer_time_after` to `None` if it is `None`, but perhaps the point here is to replace other falsy values like empty strings with `None`?

File: api/graphql/reservations/reservation_serializers/staff_create_serializers.py [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+
+        if not data.get("buffer_time_before"):
+            data["buffer_time_after"] = None
```
Comment (saaste) [inline api/graphql/reservations/reservation_serializers/staff_create_serializers.py:182]: Is this correct? Should it be `data["buffer_time_before"]`?

**Review by saaste — CHANGES_REQUESTED** [review]
Once again, amazing work with the refactoring! 👏 

Just two small questions but otherwise looks great! 👍

File: api/graphql/reservations/reservation_serializers/staff_create_serializers.py (hunk 2 above)
Comment (nicobav) [reply api/graphql/reservations/reservation_serializers/staff_create_serializers.py:182]: should but as you pointed out should not be necessary to touch it and that's why tests passes 😂

File: api/graphql/reservations/reservation_serializers/staff_create_serializers.py (hunk 1 above)
Comment (nicobav) [reply api/graphql/reservations/reservation_serializers/staff_create_serializers.py:179]: Yup, no point, I was so heavily comparing the code to the "regular" create where the buffer time gets copied from reservation_unit. I'll change this. Good catch 👍

Comment (nicobav) [reply api/graphql/reservations/reservation_serializers/staff_create_serializers.py]: its now removed 🧘‍♂️

**Review by saaste — APPROVED** [review]
Jokeri pokeri box! :shipit:

**Pull request merged**